### PR TITLE
Add more tracing for azure log

### PR DESCRIPTION
**Pull request by suwatch**
See ample in internal email title "kudu traces to Geneva".

### CODE REVIEW

File: Kudu.Core/Tracing/KuduEventSource.cs [hunk 1]
```diff
@@ -60,5 +60,23 @@ public void WebJobEvent(string siteName, string jobName, string Message, string
                 WriteEvent(65513, siteName, jobName, Message, jobType, error);
             }
         }
+
+        [Event(65514, Level = EventLevel.Warning, Message = "Generic event for site {0}", Channel = EventChannel.Operational)]
+        public void GenericEvent(string siteName, string Message, string requestId, string scmType, string siteMode, string buildVersion)
+        {
+            if (IsEnabled())
+            {
+                WriteEvent(65514, siteName, Message, requestId, scmType, siteMode, buildVersion);
```
Comment (davidebbo) [inline Kudu.Core/Tracing/KuduEventSource.cs]: Does that all go to the same existing Kudu table?

File: Kudu.Services.Web/Tracing/TraceModule.cs [hunk 2]
```diff
@@ -116,10 +121,16 @@ private static void OnBeginRequest(object sender, EventArgs e)
...
-                    !key.Equals("X-MS-CLIENT-PRINCIPAL-NAME", StringComparison.OrdinalIgnoreCase))
+                    !key.Equals("X-MS-CLIENT-PRINCIPAL-NAME", StringComparison.OrdinalIgnoreCase) &&
+                    !key.Equals(Constants.SiteRestrictedJWT, StringComparison.OrdinalIgnoreCase))
                 {
                     attribs[key] = httpContext.Request.Headers[key];
                 }
+                else
+                {
+                    var value = httpContext.Request.Headers[key];
+                    attribs[key] = string.IsNullOrEmpty(value) ? value : (value.Substring(0, Math.Min(3, value.Length)) + "...");
```
Comment (davidebbo) [inline Kudu.Services.Web/Tracing/TraceModule.cs]: This needs a comment, as it's non-obvious what it's doing

Comment (suwatch) [inline Kudu.Services.Web/Tracing/TraceModule.cs]: Fixed.

```csharp
// for sensitive headers, we only trace first 3 characters following by "..."
var value = httpContext.Request.Headers[key];
attribs[key] = string.IsNullOrEmpty(value) ? value : (value.Substring(0, Math.Min(3, value.Length)) + "...");
```

File: Kudu.Core/Tracing/KuduEventSource.cs (hunk 1 above)
Comment (suwatch) [inline Kudu.Core/Tracing/KuduEventSource.cs]: Yes all to Kudu table.  I have the sample on separate email thread (title Add kudu traces to Geneva (Kusto)).

**Pull request closed without merging**